rpm/st2chatops.spec

```
@@ -21,6 +21,11 @@ Prefix:         /opt/stackstorm/chatops
 %define _rpmdir %(pwd)/..
 %define _build_name_fmt %%{NAME}-%%{VERSION}-%%{RELEASE}.%%{ARCH}.rpm
 
+%if 0%{?rhel} >= 8
+BuildRequires:  python36-rpm-macros
+%undefine %brp_mangle_shebangs
```

Could be the following then:
```suggestion
%undefine __brp_mangle_shebangs
```

---

Let's try

---

It worked, but not `Requires` doesn't look quite right now.

Previous `rpm` required some libs:
```
Requires: /bin/bash /bin/sh /usr/bin/./node_modules/.bin/coffee /usr/bin/bash /usr/bin/coffee /usr/bin/node /usr/bin/perl /usr/bin/ruby /usr/bin/sh libc.so.6()(64bit) libc.so.6(GLIBC_2.14)(64bit) libc.so.6(GLIBC_2.2.5)(64bit) libgcc_s.so.1()(64bit) libgcc_s.so.1(GCC_3.0)(64bit) libicudata.so.60()(64bit) libicui18n.so.60()(64bit) libicuuc.so.60()(64bit) libm.so.6()(64bit) libpthread.so.0()(64bit) libpthread.so.0(GLIBC_2.2.5)(64bit) libstdc++.so.6()(64bit) libstdc++.so.6(CXXABI_1.3)(64bit) libstdc++.so.6(CXXABI_1.3.9)(64bit) libstdc++.so.6(GLIBCXX_3.4)(64bit) rtld(GNU_HASH)
```
New one:
```
Requires: /bin/bash /bin/sh /usr/bin/env
```

Doesn't have them anymore. Curious what's the diff...

---

Weird. I wonder if tweaking the node install to a current version affected things: 

https://github.com/StackStorm/st2chatops/blob/ff52f6b1dfb6ac28572153a476670f5ca0faa15e/packagingenv/centos8/Dockerfile#L9-L13

---

Don't think that's the reason.

Since `%undefine __brp_mangle_shebangs` removes all the lib requirements, let's try to exclude  python, ruby and coffee binaries in a more targeted way via `%global __requires_exclude`.

See: https://docs.fedoraproject.org/en-US/packaging-guidelines/AutoProvidesAndRequiresFiltering/#_filtering_provides_and_requires_after_scanning

---

@armab Do you mean excude `perl, ruby and coffee`? 

```
Examining /root/build/st2chatops-3.2dev-1.x86_64.rpm: st2chatops-3.2dev-1.x86_64
Marking /root/build/st2chatops-3.2dev-1.x86_64.rpm to be installed
Resolving Dependencies
--> Running transaction check
---> Package st2chatops.x86_64 0:3.2dev-1 will be installed
--> Processing Dependency: /usr/bin/./node_modules/.bin/coffee for package: st2chatops-3.2dev-1.x86_64
Loading mirror speeds from cached hostfile
 * base: us.mirror.nsec.pt
 * extras: us.mirror.nsec.pt
 * updates: us.mirror.nsec.pt
--> Processing Dependency: /usr/bin/coffee for package: st2chatops-3.2dev-1.x86_64
--> Processing Dependency: /usr/bin/perl for package: st2chatops-3.2dev-1.x86_64
--> Processing Dependency: /usr/bin/ruby for package: st2chatops-3.2dev-1.x86_64
--> Processing Dependency: libstdc++.so.6(CXXABI_1.3.9)(64bit) for package: st2chatops-3.2dev-1.x86_64
--> Processing Dependency: libicudata.so.60()(64bit) for package: st2chatops-3.2dev-1.x86_64
--> Processing Dependency: libicui18n.so.60()(64bit) for package: st2chatops-3.2dev-1.x86_64
--> Processing Dependency: libicuuc.so.60()(64bit) for package: st2chatops-3.2dev-1.x86_64
--> Running transaction check
```

---

Yes, exactly!

---

Done